Subdivision_method_3/include/CGAL/Subdivision_method_3/subdivision_masks_3.h

```diff
@@ -331,6 +331,7 @@ class Loop_mask_3 : public PQQ_stencil_3<PolygonMesh, VertexPointMap> {
   void vertex_node(vertex_descriptor vertex, Point& pt) {
     Halfedge_around_vertex_circulator vcir(vertex, *(this->pmesh));
     size_t n = circulator_size(vcir);
+    CGAL_assertion(n > 0);
```

Comment:
```suggestion
    CGAL_assume(n > 0);
```
so that the condition is known by the compiler even without assertions.

Reply:
But even with  assume, would it not make more sense to check the vertex?

Reply:
Even we add an assumption `vertex != boost::graph_traits<Mesh>::null_vertex()`, I doubt the compiler can infer from it that `circulator_size(vcir)` is not null.

Reply:
But are there not many places in our code where we divide and are just sure that it is not zero?

Reply:
The static analysis is specifically inferring that n can be zero. But I am happy to try other suggestions.

Reply:
> But are there not many places in our code where we divide and are just sure that it is not zero?

I do not know. We are only discussing this one because Giles ran static analysis on this piece of code.

@GilesBathgate Do you ran analysis on specific code (and why that one?) or on all of CGAL?

Reply:
@lrineau I run the static analysis on my project (RapCAD) https://github.com/GilesBathgate/RapCAD/runs/2461012571?check_suite_focus=true which includes parts of CGAL that I use. I could perhaps one day configure it to run on all of CGAL, but I am not sure how to set that up.

Reply:
> @lrineau I run the static analysis on my project (RapCAD) [GilesBathgate/RapCAD/runs/2461012571?check_suite_focus=true](https://github.com/GilesBathgate/RapCAD/runs/2461012571?check_suite_focus=true) which includes parts of CGAL that I use. I could perhaps one day configure it to run on all of CGAL, but I am not sure how to set that up.

Is there somewhere we can see the output of Coverity? That would be more useful than initializing the variables.

Reply:
Well it just says:

>Source code is not available because it has been purged.

Because I've applied this patch to test the fix. But essentially its saying that on the line:
`const FT Cn = (FT) (5.0/8.0 - CGAL::square(3+2*std::cos(2 * CGAL_PI/(double) n))/64.0);`
If n is 0 then:

>355591 Division or modulo by float zero
This could cause an immediate crash or incorrect computation.
In CGAL::​Loop_mask_3<CGAL::​Polyhedron_3<CGAL::​Epeck, CGAL::​Polyhedron_items_3, CGAL::​HalfedgeDS_default, std::​allocator<int>>, CGAL::​internal::​Point_accessor<CGAL::​internal::​In_place_list_iterator<CGAL::​HalfedgeDS_in_place_list_vertex<CGAL::​I_Polyhedron_vertex<CGAL::​HalfedgeDS_vertex_base<CGAL::​HalfedgeDS_list_types<CGAL::​Epeck, CGAL::​I_Polyhedron_derived_items_3<CGAL::​Polyhedron_items_3>, std::​allocator<int>>, CGAL::​Boolean_tag<true>, CGAL::​Point_3<CGAL::​Epeck>>>>, std::​allocator<CGAL::​HalfedgeDS_in_place_list_vertex<CGAL::​I_Polyhedron_vertex<CGAL::​HalfedgeDS_vertex_base<CGAL::​HalfedgeDS_list_types<CGAL::​Epeck, CGAL::​I_Polyhedron_derived_items_3<CGAL::​Polyhedron_items_3>, std::​allocator<int>>, CGAL::​Boolean_tag<true>, CGAL::​Point_3<CGAL::​Epeck>>>>>>, CGAL::​Point_3<CGAL::​Epeck>, CGAL::​Point_3<CGAL::​Epeck> &, false>>::​vertex_node(CGAL::​internal::​In_place_list_iterator<CGAL::​HalfedgeDS_in_place_list_vertex<CGAL::​I_Polyhedron_vertex<CGAL::​HalfedgeDS_vertex_base<CGAL::​HalfedgeDS_list_types<CGAL::​Epeck, CGAL::​I_Polyhedron_derived_items_3<CGAL::​Polyhedron_items_3>, std::​allocator<int>>, CGAL::​Boolean_tag<true>, CGAL::​Point_3<CGAL::​Epeck>>>>, std::​allocator<CGAL::​HalfedgeDS_in_place_list_vertex<CGAL::​I_Polyhedron_vertex<CGAL::​HalfedgeDS_vertex_base<CGAL::​HalfedgeDS_list_types<CGAL::​Epeck, CGAL::​I_Polyhedron_derived_items_3<CGAL::​Polyhedron_items_3>, std::​allocator<int>>, CGAL::​Boolean_tag<true>, CGAL::​Point_3<CGAL::​Epeck>>>>>>, CGAL::​Point_3<CGAL::​Epeck> &): An expression which may be zero is used as a divisor in floating-point arithmetic

Obviously its not clever enough to determine that n can never be zero, but since this is also an assumption that the author made I thought it would be appropriate to use CGAL_assertion, or CGAL_assume to document this in the code.

Reply:
> > @lrineau I run the static analysis on my project (RapCAD) [GilesBathgate/RapCAD/runs/2461012571?check_suite_focus=true](https://github.com/GilesBathgate/RapCAD/runs/2461012571?check_suite_focus=true) which includes parts of CGAL that I use. I could perhaps one day configure it to run on all of CGAL, but I am not sure how to set that up.
> 
> Is there somewhere we can see the output of Coverity? That would be more useful than initializing the variables.

Sorry: I mixed this PR up with another of the ones about uninitialized data.